ml-agents/mlagents/trainers/tests/test_trainer_util.py

```diff
@@ -137,3 +141,47 @@ def test_existing_directories(tmp_path):
     os.mkdir(init_path)
     # Should pass since the directory exists now.
     validate_existing_directories(output_path, False, True, init_path)
+
+
+@pytest.mark.parametrize("dir_exists", [True, False])
+def test_setup_init_path(tmpdir, dir_exists):
+    """
+
+    :return:
+    """
+    test_yaml = """
+    behaviors:
+        BigWallJump:
+            init_path: BigWallJump-6540981.pt #full path
+            trainer_type: ppo
+        MediumWallJump:
+            init_path: {}/results/test_run_id/MediumWallJump/checkpoint.pt
+            trainer_type: ppo
+        SmallWallJump:
+            trainer_type: ppo
+    checkpoint_settings:
+        run_id: test_run_id
+        initialize_from: test_run_id
+    """.format(
+        tmpdir
+    )
+    run_options = RunOptions.from_dict(yaml.safe_load(test_yaml))
+    if dir_exists:
+        init_path = tmpdir.mkdir("results").mkdir("test_run_id")
+        big = init_path.mkdir("BigWallJump").join("BigWallJump-6540981.pt")
+        big.write("content")
+        med = init_path.mkdir("MediumWallJump").join("checkpoint.pt")
+        med.write("content")
+        small = init_path.mkdir("SmallWallJump").join("checkpoint.pt")
+        small.write("content")
+
+        setup_init_path(run_options.behaviors, init_path)
+        assert run_options.behaviors["BigWallJump"].init_path == big
+        assert run_options.behaviors["MediumWallJump"].init_path == med
+        assert run_options.behaviors["SmallWallJump"].init_path == small
```

Comment:
probably should delete these files and folders after the assert call. I would also give a different name than "results" to the directory. Something like `test_setup_init_path_results`

Reply:
[pytest](https://docs.pytest.org/en/6.2.x/tmpdir.html#the-default-base-temporary-directory) deletes the stale tmp directories automatically. I'll update the directory name.

Reply:
done